### DISCUSSION TITLE
How to check if a file exists in public directory?

### DISCUSSION THREAD

**Discussion by Pawel2505** [nuxt/nuxt#16756] [Questions]
I am trying to check if a video file which is located in public directory exist or not. I try it with try-catching require(path_to_the_file) method. If an exception occurs - the file does not exist and vice versa if no exception is thrown - the file exist.
However I have no clue which path-to-the-file should be used in "require" method. 
I tried with: 
1. full URL 
2. relative file url inside public directory (which works fine for the html video element bwt. but not for "require" method.
3. adding relative prefixes to file url like: '../../'+file.mp4
4. adding prefix 'public' etc.

Nothing worked. What path should I use? Or maybe any better method to check if a file exists or not?

**Comment by sevillaarvin** [comment]
Why are you trying to do this? Maybe you’re looking for a different solution altogether.

**Reply by Pawel2505** [reply]
I just want to display video if it exists and display something else, if doesn't.

**Comment by TheAlexLichter** [comment]
Couldn't you create a `fetch` call (e.g. as `HEAD` so you don't load the whole content) to the URL of the file and check the error code?

**Reply by Pawel2505** [reply]
It could work, but it seems not a way to go for me with the error code. But maybe I am wrong. Anyway it seems to be a basic operation in a web programming language - just checking if a file on my server exists or not.

**Reply by TheAlexLichter** [reply]
Well, on the client - this is what you usually do in the browser. Request content (e.g. via HTTP) and - if it is not there - use a fallback.

You can even use plain HTML via a `<video>` tag and `onerror` to handle it.
See e.g. https://stackblitz.com/edit/https-github-comnuxtframeworkdiscussions6603?file=src%2FApp.vue
